### DISCUSSION TITLE
Enable "all" Favicons in re-learn Theme

### DISCUSSION THREAD

**Discussion by davidbeckonline** [McShelby/hugo-theme-relearn#839] [Q&A] [answered]
Hello.
I would like to consider "All" Favicons on my re-learn theme based website.

It seems that
https://realfavicongenerator.net
is a great website for this purpose and often referred to on discourse.gohugo.io . 

I would need some help to apply the code from this website in the re-learn theme.

Providing an image on the mentioned website is generating me a package of `png` files, an `xml`, and an `ico`, and a `site.webmanifest` .

Furthermore, I am getting some HTML code to be included on the head section:

```
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
```

How could I apply this within the re-learn theme?
So far, I have
1. added the files to my `static/images` directory
2. created a `layouts/partials/favicon.html` file, in which I copied the code from above

But that does not really work. I the favicon.ico is getting considered in a desktop browser, but nothing else, from what I see.

**Comment by McShelby** [comment, marked as answer]
Try something like 

````
<link rel="apple-touch-icon" sizes="180x180" href="{{ "apple-touch-icon.png" | relURL }}>
.
.
.
````

Note that I removed the leading slash!

**Reply by davidbeckonline** [reply]
Spot on. The path had to be adjusted.

In my scenario, I decided to move all the files generated by https://realfavicongenerator.net/ into directory
`.../static/images/favicon`
and then adjust the code as folllows:

```
<link rel="apple-touch-icon" sizes="180x180" href="images/favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="images/favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="images/favicon/favicon-16x16.png">
<link rel="manifest" href="images/favicon/site.webmanifest">
<link rel="mask-icon" href="images/favicon/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
```

Thank you very much for the feedback.
